File: ANUVANSH/DP_7.py
"""Here is the problem statement for August 16th
You are part of a team developing a text analysis tool called "TextInsight." 
As part of this tool, you have been assigned the task of creating a feature that allows users to analyze the frequency of characters in a given text. 
This feature will assist users in understanding the distribution of characters and identifying patterns within the text, 
which can be helpful for tasks such as language analysis, cryptography, or data validation."""

def count_frequency(text):
    """
    Analyzes the frequency of characters in a given text.

    Args:
        text (str): The input text.

    Returns:
        dict: A dictionary containing the character frequencies.
    """
    freq = {}
    for i in text:
        if i in freq:
            freq[i] += 1
        else:
            freq[i] = 1
    return freq

def main():
    """
    Main function to run the TextInsight Character Frequency Analysis feature.
    """
    print("Welcome to TextInsight Character Frequency Analysis!")

    user_text = input("Please enter the text you would like to analyze: ")

    freq_count = count_frequency(user_text)
    
    print("\nThank you for providing the text for analysis.")
    print("Here is the character frequency analysis:")
    for character, count in freq_count.items():
        if count == 1:
            print(f"- '{character}' occurs {count} time")
        else:
            print(f"- '{character}' occurs {count} times")

if __name__ == "__main__":
    main()

File: Dipti/DP-7.js
/*You are part of a team developing a text analysis tool called "TextInsight." As part of this tool, you have been 
assigned the task of creating a feature that allows users to analyze the frequency of characters in a given text.
 This feature will assist users in understanding the distribution of characters and identifying patterns within the text,
 which can be helpful for tasks such as language analysis, cryptography, or data validation. */


function analyzeCharacterFrequency(text) {
    /* function for analyze frequency of characters.
    args:
        text = text entered by user.
    returns:
        return the frequency count of the character. */
    
    const frequencyMap = new Map();

    // Remove spaces and convert text to lowercase for case-insensitive analysis
    const cleanedText = text.replace(/\s/g, '').toLowerCase();

    // Count character occurrences
    for (const char of cleanedText) {
        if (frequencyMap.has(char)) {
            frequencyMap.set(char, frequencyMap.get(char) + 1);
        } else {
            frequencyMap.set(char, 1);
        }
    }

    return frequencyMap;
}

function displayCharacterFrequencyAnalysis(frequencyMap) {
    /* function for display the frequency of character.
    args:
        frequency map return by analyzeCharacterFrequency function.
    returns:
        no return. */
    
    console.log("Here is the character frequency analysis:");
    
    frequencyMap.forEach((count, char) => {
        console.log(`- '${char}' occurs ${count} time${count > 1 ? 's' : ''}`);
    });
}


const prompt = require("prompt-sync")({sigint: true});
console.log("Welcome to TextInsight Character Frequency Analysis!");
const text = prompt("Please enter the text you would like to analyze:");
const frequencyMap = analyzeCharacterFrequency(text);
console.log(`Thank you for providing the text for analysis.`);
displayCharacterFrequencyAnalysis(frequencyMap);

